finality-aleph/src/sync/mock/backend.rs

```diff
@@ -352,3 +361,42 @@ impl ChainStatus<MockBlock, MockJustification> for Backend {
         }
     }
 }
+
+#[derive(Debug)]
+pub enum VerifierError {
+    IncorrectJustification,
+    IncorrectSession,
+}
+
+impl Display for VerifierError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), FmtError> {
+        write!(f, "{:?}", self)
+    }
+}
+
+impl Verifier<MockJustification> for Backend {
+    type Error = VerifierError;
+
+    fn verify(
+        &mut self,
+        justification: MockJustification,
+    ) -> Result<MockJustification, Self::Error> {
+        let top_number = self.top_finalized().unwrap().header.id.number;
+        let storage = self.inner.lock();
+        let current_session = storage
+            .session_boundary_info
+            .session_id_from_block_num(top_number);
+        let justification_session = storage
+            .session_boundary_info
+            .session_id_from_block_num(justification.id().number);
+        if !(justification_session == current_session
+            || justification_session == current_session.next())
+        {
+            return Err(Self::Error::IncorrectSession);
```

Comment:
Why cant you just verify if the justification is correct for older sessions?

Reply:
I think it might be too restrictive in test scenarios. But some wiggle room with cache size would be better here :+1:

Reply:
but test scenarios should reflect the real case...

Reply:
Lets tie this to the `VERIFIER_CACHE_SIZE ` :)

Reply:
done